scan/lib/scan/runner.rb

```diff
@@ -80,8 +84,16 @@ def handle_results(tests_exit_status)
     end
 
     def test_results
-      return "" unless Scan.cache[:temp_junit_report]
-      File.read(Scan.cache[:temp_junit_report])
+      temp_junit_report = Scan.cache[:temp_junit_report]
+      return File.read(temp_junit_report) if temp_junit_report && File.file?(temp_junit_report)
```

Comment:
1. `test_results` is only called once per run of scan, so I think that is fine.
2. We believe the garbage collection is causing the tempfile to be deleted before this code is hit.
3. There really isn't a reason, I just thought that writing out the whole `cat #{file} | xcpretty --report junit --output ...` was much more straightforward.